src/rules/getTypeExports.ts

```diff
@@ -25,27 +38,40 @@ function isType(
  * all of the exported Types/Interfaces. The exported interfaces and types will
  * be stored in a cache.
  */
-function parseTSTreeForExportedTypes(cacheKey: string, content: string): void {
+function parseTSTreeForExportedTypes(
+  cacheKey: string,
+  content: string,
+  context: LintContext,
+): void {
   const typeList: string[] = [];
-
   try {
     if (fileCache && fileCache.get(cacheKey)) {
       const ast = parse(content, { sourceType: 'module' });
       const cache = fileCache.get(cacheKey);
-
       if (!cache) {
         return;
       }
 
       ast.body.forEach((node: TSESTree.Statement) => {
         if (node.type === 'ExportNamedDeclaration') {
+          let typeExports: Set<string> | undefined = new Set();
+          const { source } = node;
+          const sourceName =
+            source && 'value' in source ? source.value : undefined;
+          if (typeof sourceName === 'string') {
+            // eslint-disable-next-line @typescript-eslint/no-use-before-define
+            typeExports = parseFileForTypedExports(sourceName, context);
+          }
```

Comment:
This recursively traverses the AST to get types that have been immediately exported through mutliple layers of modules. Since the caching is in use, it's effectively the same speed but means you can infer types of no type information is present in the next block.